code/modules/admin/admin_verbs.dm

```diff
@@ -534,7 +534,7 @@
 	set name = "dsay"
 	set hidden = TRUE
 
-	if(!check_rights(R_ADMIN|R_MENTOR))
+	if(!msg || !check_rights(R_ADMIN|R_MENTOR))
 		return
```

Comment:
Why not just move the below checks to under the !msg check?

Reply:
!msg check happens much later after a bunch of other branching stuff and a spam check plus a sanitize that just don't ever need to run if we don't provide a dirty msg to begin with

Reply:
yes but the checks don't matter when they are runs since it all goes to the same anyway difference is that it fails silently instead

Reply:
the question is do we care about trying to sanitize the empty string that we would get from a blank input or cancel button? if the perf hit is nothing than we can just do it first or maybe right after the `check_rights()` if we want to skip the sanitize for blank msg we need to check its dirty contents somewhere

Reply:
i mean if you're considering perf you check message twice, and this is not exactly the hardest hitting part of speech